sdks/java/core/src/main/java/org/apache/beam/sdk/values/Row.java

```diff
@@ -422,14 +422,8 @@ static int deepHashCode(Object a, Schema.FieldType fieldType) {
         V value = e.getValue();
         V otherValue = b.get(key);
 
-        if (value == null) {
-          if (otherValue != null || !b.containsKey(key)) {
-            return false;
-          }
-        } else {
-          if (!deepEquals(value, otherValue, valueType)) {
-            return false;
-          }
+        if (!deepEquals(value, otherValue, valueType)) {
```

Comment:
This won't be right if maps allow for null values. For example, if we have `a = { "s": null }` and `b = { }` then you need the `containsKey` check.

Reply:
Right, I think `a = {"s": null}` and `b = {"t": null}` would be better example as they need to have the same size first

Reply:
Would you like to add a test for this case so that future similar changes which would break this would be caught?

Reply:
`RowTest#testCreateAndCompareNullableMapWithDifferentKey` covers this case, what do you think?